presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorPartitioningMetadata.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.connector;
+
+import java.util.List;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorPartitioningMetadata
+{
+    private final ConnectorPartitioningHandle partitioningHandle;
+    private final List<String> partitionColumns;
```

Comment:
I am wondering, do we also want to make `ConnectorPartitioningMetadata` to be able to model constant values? (basically a weaker version of `ArgumentBinding`)

I am thinking about the use case that we want to materialize exchanges including constant / replicated null. Hive connector only supports bucket on columns. But imaging custom connector (e.g. MemoryConnector or our own ExchangeConnector) could support constant participating in bucket function. 

Just a thought. We can always add them later (and we can have more individual tasks to ask other engineers to work on :) )

Reply:
I was also thinking about something like that. But frankly i don't want to introduce SPI without an example of a real usage of it. It is very easy to over engineer something that you don't know how to use, or just go into a wrong direction.

So let's keep it simple for now. We can extend it later. The temporary table SPI is experimental anyway.

Reply:
> So let's keep it simple for now. We can extend it later. The temporary table SPI is experimental anyway.

I agree. Especially given the whole Materialize Exchange is not simple anyway. But it doesn't hurt to write down these random thoughts so we can improve it later :)